src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -512,9 +512,12 @@ private void ResolveAuthority()
                     Config.AuthorityInfo.AuthorityType == AuthorityType.Aad)
                 {
                     AadAuthority aadAuthority = Authority.CreateAuthority(Config.AuthorityInfo) as AadAuthority;
-                    if (!aadAuthority.IsCommonOrganizationsOrConsumersTenant())
+                    if (!aadAuthority.IsCommonOrganizationsOrConsumersTenant() && 
```

Comment:
B2C will work as well, correct?

Reply:
yes, this whole logic is for when `Config.AuthorityInfo.AuthorityType == AuthorityType.Aad`

I guess we could throw a meaningful exception if authority is B2C and "WithTenant" is used. I think we do nothing right now.